python/Ganga/GPIDev/Base/Objects.py

```diff
@@ -365,22 +365,24 @@ def __get__(self, obj, cls):
             return cls._schema[name]
 
         if self._getter_name:
-            return self._bind_method(obj, self._getter_name)()
+            returnable = self._bind_method(obj, self._getter_name)()
+            if isinstance(returnable, GangaObject):
+                returnable._setParent(self)
+            return returnable
```

Comment:
The problem is with your code which sets the parent in `__get__`, it overwrites any existing parent which is there already. So objects which had correct parents are getting their parents changed to something wrong. At least if we remove this code then and existing parents will remain which up until now has worked fine.

The code you added to `__get__` here is only needed is the getter is creating a new object and returning it parentless which I guess does not happen very often.